### PR TITLE
feat(ui): support auto login for the demo

### PR DESCRIPTION
**Pull request by wwayne**
Screen record: https://jam.dev/c/a3f3d6b6-57f7-4ba8-a33e-465b17b6c01b

The auto-login state is stored in SessionStorage.
Auto-login occurs only in a new tab
Otherwise, it simply autofills the fields.

### CODE REVIEW

File: ee/tabby-ui/app/auth/signin/components/user-signin-form.tsx [hunk 1]
```diff
@@ -42,15 +45,37 @@ interface UserAuthFormProps extends React.HTMLAttributes<HTMLDivElement> {
...
 }: UserAuthFormProps) {
   const isEmailConfigured = useIsEmailConfigured()
+  const isDemoMode = useIsDemoMode()
   const form = useForm<z.infer<typeof formSchema>>({
     resolver: zodResolver(formSchema)
   })
+  const formRef = React.useRef<HTMLFormElement | null>(null)
+
+  React.useEffect(() => {
+    const storageKey = 'isAutoLoggedIn'
```
Comment (wsxiaoys) [inline ee/tabby-ui/app/auth/signin/components/user-signin-form.tsx:66]:
```suggestion
    const storageKey = '_tabby_demo_autologin'
```

also put this as a global var

Comment (wwayne) [reply ee/tabby-ui/app/auth/signin/components/user-signin-form.tsx]: Added into `lib/constants/index.ts`

File: ee/tabby-ui/lib/constants/index.ts [hunk 2]
```diff
@@ -1,3 +1,7 @@
 export const PLACEHOLDER_EMAIL_FORM = 'name@yourcompany.com'
 
 export const DEFAULT_PAGE_SIZE = 20
+
+export const STORAGE_KEY = {
```
Comment (wsxiaoys) [inline ee/tabby-ui/lib/constants/index.ts:5]:
```suggestion
export const SESSION_STORAGE_KEY = {
```

**Review by wsxiaoys — APPROVED** [review]

File: ee/tabby-ui/lib/constants/index.ts (hunk 2 above)
Comment (wwayne) [reply ee/tabby-ui/lib/constants/index.ts]: renamed

**Review by liangfung — APPROVED** [review]

**Pull request merged**